bigchaindb/common/transaction.py

```diff
@@ -812,13 +813,17 @@ def _sign_simple_signature_fulfillment(cls, input_, message, key_pairs):
         #       this should never happen, but then again, never say never.
         input_ = deepcopy(input_)
         public_key = input_.owners_before[0]
+        if input_.fulfills:
+            message = '{}{}{}'.format(
```

Comment:
Suggestion: here, and in other places in this PR, you can also leverage the method [`hash.update`](https://docs.python.org/3/library/hashlib.html#hashlib.hash.update), it's actually quite nice:

```python
my_sha3_obj = sha3_256()

if input_.fulfills:
    my_sha3_obj.update(message)
    my_sha3_obj.update(input_.fulfills.txid)
    my_sha3_obj.update(input_.fulfills.output)

message = my_sha3_obj.digest()
```

(You might need to encode the strings to bytes, so the real code might look a bit uglier.)

Reply:
@vrde Implemented a variation of your suggestion in 1eecfc3 

If that is okay with you we can then move forward with merging the PR.

Reply:
My idea was to avoid the `'{}{}'.format(a, b)` entirely, and replace it with consecutive `message.update(a)` and `message.update(b)`, but now the approach is mixed. It's not that important tho :)